### PR TITLE
Auto-populate the NGR value based on address lookup result

### PR DESCRIPTION
**Pull request by autotelik**
WAS-1022 - Auto-populate the NGR value based on address lookup result

new conversion module; class to perform xy to os grid reference conversions

### CODE REVIEW

File: lib/digital_services_core/services/address_lookup/conversion/xy_to_os_national_grid_ref.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Specific converter - convert EA AddressFacade into other usable forms
+#
+module DigitalServicesCore
+  module Services
+    module AddressLookup
+      module Conversion
+
+        class XyToOsNationalGridRef
+
+          def convert(location)
```
Comment (rickclare) [inline lib/digital_services_core/services/address_lookup/conversion/xy_to_os_national_grid_ref.rb]: Can we add a comment linking to the article or post that we used as the basis of our "conversion algorithm" ?

**Comment by rickclare** [comment]
Apart from the above comment, this is ready to merge @autotelik

File: lib/digital_services_core/services/address_lookup/conversion/xy_to_os_national_grid_ref.rb (hunk 1 above)
Comment (autotelik) [inline lib/digital_services_core/services/address_lookup/conversion/xy_to_os_national_grid_ref.rb]: currently in Alan's Evernote so I have copied to this file for now

File: app/jobs/digital_services_core/uprn_address_populator_job.rb [hunk 2]
```diff
@@ -21,7 +21,12 @@ def perform(location)
         # use to construct an Address
         address_data = parser.address_data inbound
 
-        location.address.update!(address_data[0]) if address_data.try(:any?)
+        if address_data.try(:any?)
```
Comment (rickclare) [inline app/jobs/digital_services_core/uprn_address_populator_job.rb]: @autotelik I can't think of a better way of doing this either!  If it works, I would say go ahead with this solution.

**Pull request merged**